### PR TITLE
Make checkodesol work for series solutions

### LINKED ISSUE #16000: No equivalent of checkodesol for series solutions of ODEs
Some of the ODE solvers in `dsolve` give series solutions. Other types of solutions when listed in the tests are checked with either `checkodesol` or `checksysodesol`. There is no equivalent for series solutions and `checkodesol` doesn't work:
```julia
In [16]: eq = f(x).diff(x) - f(x)                                                                                                                             

In [17]: dsolve(eq, hint='1st_power_series')                                                                                                                  
Out[17]: 
                       2       3       4       5        
                   C₁⋅x    C₁⋅x    C₁⋅x    C₁⋅x     ⎛ 6⎞
f(x) = C₁ + C₁⋅x + ───── + ───── + ───── + ───── + O⎝x ⎠
                     2       6       24     120         

In [18]: eq = f(x).diff(x) - f(x)                                                                                                                             

In [19]: sol = dsolve(eq, hint='1st_power_series')                                                                                                            

In [20]: checkodesol(eq, sol)                                                                                                                                 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/current/sympy/sympy/sympy/core/cache.py in wrapper(*args, **kwargs)
     93                 try:
---> 94                     retval = cfunc(*args, **kwargs)
     95                 except TypeError:

~/current/sympy/sympy/sympy/series/order.py in __new__(cls, expr, *args, **kwargs)
    151         if not all(v.is_symbol for v in variables):
--> 152             raise TypeError('Variables are not symbols, got %s' % variables)
    153 

TypeError: Variables are not symbols, got [1, 0]

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-20-48cca9015ba5> in <module>
----> 1 checkodesol(eq, sol)

~/current/sympy/sympy/sympy/solvers/ode.py in checkodesol(ode, sol, func, order, solve_for_func)
   2505                     rhs = sub_func_doit(rhs, func.diff(x, i), diffsols[i])
   2506                     ode_or_bool = Eq(lhs, rhs)
-> 2507                     ode_or_bool = simplify(ode_or_bool)
   2508 
   2509                     if isinstance(ode_or_bool, (bool, BooleanAtom)):

~/current/sympy/sympy/sympy/simplify/simplify.py in simplify(expr, ratio, measure, rational, inverse)
    515 
    516     try:
--> 517         return expr._eval_simplify(ratio=ratio, measure=measure, rational=rational, inverse=inverse)
    518     except AttributeError:
    519         pass

~/current/sympy/sympy/sympy/core/relational.py in _eval_simplify(self, ratio, measure, rational, inverse)
    468         # standard simplify
    469         e = super(Equality, self)._eval_simplify(
--> 470             ratio, measure, rational, inverse)
    471         if not isinstance(e, Equality):
    472             return e

~/current/sympy/sympy/sympy/core/relational.py in _eval_simplify(self, ratio, measure, rational, inverse)
    246             if dif.is_comparable:
    247                 v = dif.n(2)
--> 248             elif dif.equals(0):  # XXX this is expensive
    249                 v = S.Zero
    250             if v is not None:

~/current/sympy/sympy/sympy/core/expr.py in equals(self, other, failing_expression)
    702             return False
    703 
--> 704         constant = diff.is_constant(simplify=False, failing_number=True)
    705 
    706         if constant is False:

~/current/sympy/sympy/sympy/core/expr.py in is_constant(self, *wrt, **flags)
    618             try:
    619                 a = expr.subs(list(zip(free, [0]*len(free))),
--> 620                     simultaneous=True)
    621                 if a is S.NaN:
    622                     # evaluation may succeed when substitution fails

~/current/sympy/sympy/sympy/core/basic.py in subs(self, *args, **kwargs)
    990                 reps[d] = new
    991             reps[m] = S.One  # get rid of m
--> 992             return rv.xreplace(reps)
    993         else:
    994             rv = self

~/current/sympy/sympy/sympy/core/basic.py in xreplace(self, rule)
   1179 
   1180         """
-> 1181         value, _ = self._xreplace(rule)
   1182         return value
   1183 

~/current/sympy/sympy/sympy/core/basic.py in _xreplace(self, rule)
   1193             for a in self.args:
   1194                 try:
-> 1195                     a_xr = a._xreplace(rule)
   1196                     args.append(a_xr[0])
   1197                     changed |= a_xr[1]

~/current/sympy/sympy/sympy/core/basic.py in _xreplace(self, rule)
   1200             args = tuple(args)
   1201             if changed:
-> 1202                 return self.func(*args), True
   1203         return self, False
   1204 

~/current/sympy/sympy/sympy/core/cache.py in wrapper(*args, **kwargs)
     94                     retval = cfunc(*args, **kwargs)
     95                 except TypeError:
---> 96                     retval = func(*args, **kwargs)
     97                 return retval
     98 

~/current/sympy/sympy/sympy/series/order.py in __new__(cls, expr, *args, **kwargs)
    150 
    151         if not all(v.is_symbol for v in variables):
--> 152             raise TypeError('Variables are not symbols, got %s' % variables)
    153 
    154         if len(list(uniq(variables))) != len(variables):

TypeError: Variables are not symbols, got [1, 0]
```

### PR DESCRIPTION
**Pull request by oscarbenjamin**
#### References to other Issues or PRs

Fixes #16000 

#### Brief description of what is fixed or changed

Makes checkodesol work for series solutions of ODEs as returned by dsolve. I'm not sure this works for series solutions around non-zero points because dsolve doesn't yet return correct series solution around nonzero points.

#### Other comments

I'm still working on this...

#### Release Notes

* solvers
    * checkodesol now works for series solutions of ODEs.

### CODE REVIEW

File: sympy/solvers/tests/test_ode.py [hunk 1]
```diff
@@ -3076,66 +3081,84 @@ def test_issue_7081():
...
+    sol = Eq(f(x), C2*(x**3/6 + 1) + C1*x*(x**3/12 + 1) + O(x**6))
+    assert dsolve(eq) == sol
+    assert checkodesol(eq, sol) == (True, 0)
+
+    sol = Eq(f(x), C2*((x + 2)**4/6 + (x + 2)**3/6 - (x + 2)**2 + 1)
         + C1*(x + (x + 2)**4/12 - (x + 2)**3/3 + S(2))
         + O(x**6))
-    assert dsolve(eq, n=2) == Eq(f(x), C2*x + C1 + O(x**2))
+    assert dsolve(eq, x0=-2) == sol
+    # FIXME: Solution should be O((x+2)**6)
```
Comment (RituRajSingh878) [inline sympy/solvers/tests/test_ode.py]: `O((x+2)**6)` has a different meaning. So we can't get from the solution and also O(x**6) is the correct way of representing the solution. 
```
In [9]: Order((x+2)**6)
Out[9]: O(1)

```

For the` checkodesol` We can pass the argument `x0` and then we can use it for your work.

Comment (oscarbenjamin) [reply sympy/solvers/tests/test_ode.py]: What I meant is:
```julia
In [1]: O((x+2)**6, (x, -2))                                                                                                      
Out[1]: 
 ⎛       6        ⎞
O⎝(x + 2) ; x → -2⎠
```

Comment (oscarbenjamin) [reply sympy/solvers/tests/test_ode.py]: Making `dsolve` return that fails because `constantsimp` calls `cse` which absorbs all the terms #17587

**Comment by oscarbenjamin** [comment]
I'll merge this soon if no one objects

**Pull request merged**